src/objective_types.jl

```diff
@@ -12,17 +14,26 @@ function fix_order(storage_input, x_input, fun!, fun!_msg)
     end
 end
 # Used for objectives and solvers where no gradient is available/exists
-type NonDifferentiable{T} <: AbstractObjective
+type NonDifferentiable{T<:Real} <: AbstractObjective
```

Comment:
Some types might have all the necessary methods but might not be `<:Real`, and so cannot be paased